libmaven/mzSample.h

```diff
@@ -1081,7 +1082,7 @@ class PeakGroup {
         Fragment fragmentationPattern;
 
         //Issue 546: Attach hypotheses to group, same way as vector<Peak> works
-        vector<pair<Compound*, FragmentationMatchScore>> compounds{};
+        vector<pair<CompoundIon, FragmentationMatchScore>> compounds{};
 
```

Comment:
By using `CompoundIon` class instead of `Compound*`, able to decouple the `Adduct*` associated with a `Compound*` in `Compound->adductString` with some other associated `Adduct*`.